From a15b9033dcc7051951ba3b0f46782619f2b3b100 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 12 Nov 2018 14:44:17 +0800
Subject: [PATCH 5094/5242] MLK-20324 gpu: imx: dpu: tcon: Initialize tmp_m in
 tcon_cfg_videomode()

commit  3ec344d6fa0920468d4859a5c6a14467d33a8dda from
https://source.codeaurora.org/external/imx/linux-imx.git

The video mode tmp_m, as a local variable in tcon_cfg_videomode(),
is uninitialized and used to store a copy instance from the real
video mode.  tcon_cfg_videomode() would change the timing of it if
side_by_side mode is enabled.  Theoretically, there should be no
problem even if we don't initialize tmp_m.  However, coverity
reports that tmp_m is an uninitialized scalar variable when the copy
is being done(CID 5233067: Uninitialized scalar variable (UNINIT)).
This patch fixes the issue reported by coverity.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-tcon.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu/dpu-tcon.c b/drivers/gpu/imx/dpu/dpu-tcon.c
index 1e0acef..b9f15c1 100644
--- a/drivers/gpu/imx/dpu/dpu-tcon.c
+++ b/drivers/gpu/imx/dpu/dpu-tcon.c
@@ -128,7 +128,7 @@ void tcon_set_operation_mode(struct dpu_tcon *tcon)
 void tcon_cfg_videomode(struct dpu_tcon *tcon,
 			struct drm_display_mode *m, bool side_by_side)
 {
-	struct drm_display_mode tmp_m;
+	struct drm_display_mode tmp_m = { 0 };
 	struct dpu_soc *dpu = tcon->dpu;
 	const struct dpu_devtype *devtype = dpu->devtype;
 	u32 val;
-- 
1.7.9.5

